src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -561,24 +599,41 @@ public void calculateContingencySensitivityValues(PropagatedContingency continge
                                                       DcLoadFlowEngine dcLoadFlowEngine, SensitivityFactorHolder factorHolder, List<ParticipatingElement> participatingElements,
                                                       Collection<LfBus> disabledBuses, Collection<LfBranch> disabledBranches, Reporter reporter) {
         List<LfSensitivityFactor> factors = factorHolder.getFactorsForContingency(contingency.getContingency().getId());
-        if (contingency.getHvdcIdsToOpen().isEmpty()) {
-            calculateSensitivityValues(factors, factorStates, contingenciesStates, flowStates, contingencyElements,
-                    contingency, valueWriter);
-        } else { // if we have a contingency including the loss of a DC line
-            Map<LfBus, BusState> busStates = applyHvdcLineContingency(network, lfNetwork, contingency);
-            boolean shouldChangeRhs = lfParameters.isDistributedSlack()
-                    && (lfParameters.getBalanceType() == LoadFlowParameters.BalanceType.PROPORTIONAL_TO_LOAD || lfParameters.getBalanceType() == LoadFlowParameters.BalanceType.PROPORTIONAL_TO_CONFORM_LOAD);
-            DenseMatrix statesWithoutDcLine = factorStates;
-            if (shouldChangeRhs) {
-                statesWithoutDcLine = calculateStates(lfNetwork, j, equationSystem, factorGroups, lfParameters, lfParametersExt);
-            }
-            DenseMatrix flowStatesWithoutDcLine = setReferenceActivePowerFlows(dcLoadFlowEngine, equationSystem, j, factors, lfParameters, participatingElements, disabledBuses, disabledBranches, reporter);
-            calculateSensitivityValues(factors, statesWithoutDcLine, contingenciesStates, flowStatesWithoutDcLine, contingencyElements,
-                    contingency, valueWriter);
-            BusState.restoreBusStates(busStates);
-            if (shouldChangeRhs) {
-                setBaseCaseSensitivityValues(factorGroups, factorStates);
-            }
+        List<ParticipatingElement> saveParticipatingElements = new ArrayList<>(participatingElements);
```

Comment:
I suspect a serious performance regression, we should have something like before:
```java
if (contingency.getHvdcIdsToOpen().isEmpty() || contingency.getGeneratorIdsToLose().isEmpty()) { 
    calculateSensitivityValues(factors, factorStates, contingenciesStates, flowStates, contingencyElements,
                    contingency, valueWriter);
} else { // if we have a contingency including the loss of a DC line or a generator loss

}
```